src/string.cr

```diff
@@ -2040,7 +2042,7 @@ class String
   # "hello".gsub { |char| char + 1 } # => "ifmmp"
   # "hello".gsub { "hi" }            # => "hihihihihi"
   # ```
-  def gsub(&block : Char -> _)
+  def gsub(&block : Char -> Char | String) : String
```

Comment:
The return type of this block can actually be anything with a `.to_s` method. Please leave the block type restriction as-is.